code/modules/mob/living/carbon/alien/alien.dm

```diff
@@ -131,6 +131,11 @@ Des: Removes all infected images from the alien.
 		mind.name = new_xeno.real_name
 		mind.transfer_to(new_xeno)
 
+	var/obj/item/organ/internal/stomach/alien/melting_pot = get_organ_slot(ORGAN_SLOT_STOMACH)
+	var/obj/item/organ/internal/stomach/alien/frying_pan = new_xeno.get_organ_slot(ORGAN_SLOT_STOMACH)
+	if(istype(melting_pot) && istype(frying_pan))
+		while (length(melting_pot.stomach_contents))
+			frying_pan.consume_thing(melting_pot.stomach_contents[1])
```

Comment:
A runtime in consume_thing() or just a difference in organ capabilities could easily cause an infinite loop here. It's dangerous to use a while loop like this when not all the code called inside is well known ahead of time. You can use a for loop just fine, anything that doesn't move over after attempting to consume it once will just end up getting left behind like before.